From eca9ffcdc2a670a88b8c62b884816c0fe80b842e Mon Sep 17 00:00:00 2001
From: Dung Cao <dung@os.amperecomputing.com>
Date: Wed, 3 Feb 2021 07:14:23 +0000
Subject: [PATCH] Support Get System Interface Capabilities Command

Signed-off-by: Dung Cao <dung@os.amperecomputing.com>
---
 apphandler.cpp | 36 ++++++++++++++++++++++++++++++++++++
 apphandler.hpp |  1 +
 2 files changed, 37 insertions(+)

diff --git a/apphandler.cpp b/apphandler.cpp
index 90818a9..ec81b6d 100644
--- a/apphandler.cpp
+++ b/apphandler.cpp
@@ -786,6 +786,37 @@ auto ipmiAppGetBtCapabilities()
                                  outputBufferSize, transactionTime, nrRetries);
 }
 
+auto ipmiAppGetSiCapabilities(uint8_t sysIfcType)
+    -> ipmi::RspType<uint8_t, uint8_t, uint8_t, uint8_t>
+{
+    uint8_t reserved = 0x00;
+    uint8_t tranSupport;
+    uint8_t inputMsgSize;
+    uint8_t outputMsgSize;
+    switch (sysIfcType & 0xF)
+    {
+        case 0:
+            // SSIF
+            // multi-part read/write supported. Start, Middle, and End
+            // transactions support. PEC support.
+            tranSupport = 0x88;
+            inputMsgSize = 0xff;
+            outputMsgSize = 0xff -3; //minus 3 is to drop len, netfn, lun, cmd
+            break;
+        case 1:
+        case 2:
+            //KCS or SMIC
+            tranSupport = 0x00;
+            inputMsgSize = 0xff;
+            break;
+        default:
+            return ipmi::responseInvalidFieldRequest();
+    }
+
+    return ipmi::responseSuccess(reserved, tranSupport, inputMsgSize,
+                                 outputMsgSize);
+}
+
 auto ipmiAppGetSystemGuid() -> ipmi::RspType<std::array<uint8_t, 16>>
 {
     static constexpr auto bmcInterface =
@@ -1741,6 +1772,11 @@ void register_netfn_app_functions()
                           ipmi::app::cmdGetChannelCipherSuites,
                           ipmi::Privilege::None, getChannelCipherSuites);
 
+    // <Get System Interface Capabilities>
+    ipmi::registerHandler(ipmi::prioOpenBmcBase, ipmi::netFnApp,
+                          ipmi::app::cmdGetSystemIfCapabilities,
+                          ipmi::Privilege::User, ipmiAppGetSiCapabilities);
+
     // <Get System Info Command>
     ipmi::registerHandler(ipmi::prioOpenBmcBase, ipmi::netFnApp,
                           ipmi::app::cmdGetSystemInfoParameters,
diff --git a/apphandler.hpp b/apphandler.hpp
index 4f03121..53d5450 100644
--- a/apphandler.hpp
+++ b/apphandler.hpp
@@ -19,6 +19,7 @@ enum ipmi_netfn_app_cmds
     IPMI_CMD_SET_CHAN_ACCESS = 0x40,
     IPMI_CMD_GET_CHANNEL_ACCESS = 0x41,
     IPMI_CMD_GET_CHAN_INFO = 0x42,
+    IPMI_CMD_GET_SYS_IFC_CAP = 0x57,
     IPMI_CMD_SET_SYSTEM_INFO = 0x58,
     IPMI_CMD_GET_SYSTEM_INFO = 0x59,
 };
-- 
2.17.1

